src/rust/inetstack/mod.rs

```diff
@@ -339,6 +339,43 @@ impl InetStack {
         }
     }
 
+    ///
+    /// **Brief**
+    ///
+    /// Asynchronously closes a connection referred to by `qd`.
+    ///
+    /// **Return Value**
+    ///
+    /// Upon successful completion, `Ok(())` is returned. This qtoken can be used to wait until the close
+    /// completes shutting down the connection. Upon failure, `Fail` is returned instead.
+    ///
+    pub fn async_close(&mut self, qd: QDesc) -> Result<QToken, Fail> {
+        #[cfg(feature = "profiler")]
+        timer!("inetstack::async_close");
+        trace!("async_close(): qd={:?}", qd);
+
+        let future: FutureOperation = match self.lookup_qtype(&qd) {
+            Some(QType::TcpSocket) => {
+                let fut = self.ipv4.tcp.do_async_close(qd)?;
+                FutureOperation::from(fut)
+            },
+            Some(QType::UdpSocket) => {
+                let udp_op = UdpOperation::Pushto(qd, self.ipv4.udp.do_close(qd));
```

Comment:
In order to keep a symmetric design, don't we want a `self.ipv4.udp.do_async_close()`?

Reply:
This is symmetric with how UDP handles Push, where there is nothing asynchronous to do, so I left it without an additional implementation. (In fact, I forgot to change the name of the UdpOperation!)